### PR TITLE
surge: 1.8.1 -> 1.9.0

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/surge/default.nix [hunk 1]
```diff
@@ -1,45 +1,50 @@
-{ lib, stdenv, fetchFromGitHub, cmake, git, pkg-config, python3
+{ stdenv, fetchurl, fetchFromGitHub, cmake, git, pkg-config, python3
```
Comment (jonringer) [inline pkgs/applications/audio/surge/default.nix:1]:
```suggestion
{ stdenv, lib, fetchurl, fetchFromGitHub, cmake, git, pkg-config, python3
```

File: pkgs/applications/audio/surge/default.nix [hunk 2]
```diff
@@ -1,45 +1,50 @@
...
   doInstallCheck = true;
   installCheckPhase = ''
-    cd ..
+    export HOME=$(mktemp -d)
+    export SURGE_DISABLE_NETWORK_TESTS=TRUE
     build/surge-headless
   '';
 
-  meta = with lib; {
+  meta = with stdenv.lib; {
```
Comment (jonringer) [inline pkgs/applications/audio/surge/default.nix:47]:
```suggestion
  meta = with lib; {
```

Comment (jonringer) [reply pkgs/applications/audio/surge/default.nix:47]:
```
["trace: Warning: `stdenv.lib` is deprecated and will be removed in the next release. Please use `lib` instead. For more information see https://github.com/NixOS/nixpkgs/issues/108938"]
```

File: pkgs/applications/audio/surge/default.nix [hunk 3]
```diff
@@ -1,41 +1,46 @@
...
+  buildInputs = [ cairo libsndfile libxcb libxkbcommon xcbutil xcbutilcursor xcbutilkeysyms zenity curl rsync ];
 
   postPatch = ''
     substituteInPlace src/common/SurgeStorage.cpp --replace "/usr/share/Surge" "$out/share/surge"
     substituteInPlace src/linux/UserInteractionsLinux.cpp --replace '"zenity' '"${zenity}/bin/zenity'
-    substituteInPlace vstgui.surge/vstgui/lib/platform/linux/x11fileselector.cpp --replace /usr/bin/zenity ${zenity}/bin/zenity
-    patchShebangs scripts/linux/emit-vector-piggy
-    patchShebangs scripts/linux/generate-lv2-ttl
+    patchShebangs scripts/linux/
+    # cp -r $extraContent/Skins/ resources/data/skins
```
Comment (jonringer) [inline pkgs/applications/audio/surge/default.nix:31]: this is commented out, is it actually needed?

Comment (magnetophon) [reply pkgs/applications/audio/surge/default.nix]: Good catch.
It's back now.

File: pkgs/applications/audio/surge/default.nix [hunk 4]
```diff
@@ -1,41 +1,46 @@
...
     substituteInPlace src/linux/UserInteractionsLinux.cpp --replace '"zenity' '"${zenity}/bin/zenity'
-    substituteInPlace vstgui.surge/vstgui/lib/platform/linux/x11fileselector.cpp --replace /usr/bin/zenity ${zenity}/bin/zenity
-    patchShebangs scripts/linux/emit-vector-piggy
-    patchShebangs scripts/linux/generate-lv2-ttl
+    patchShebangs scripts/linux/
+    cp -r $extraContent/Skins/ resources/data/skins
   '';
 
+
   installPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/surge/default.nix:35]:
```suggestion
  preInstall = ''
```

File: pkgs/applications/audio/surge/default.nix [hunk 5]
```diff
@@ -1,41 +1,46 @@
...
-    mkdir -p $out/lib/lv2 $out/lib/vst3 $out/share/surge
-    cp -r surge_products/Surge.lv2 $out/lib/lv2/
-    cp -r surge_products/Surge.vst3 $out/lib/vst3/
-    cp -r ../resources/data/* $out/share/surge/
+    cd ..
+    cmake --build build --config Release --target install-everything-global
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/surge/default.nix:37]:
```suggestion
```

Comment (magnetophon) [reply pkgs/applications/audio/surge/default.nix:37]: There is no install target:
```
installing
install flags: SHELL=/nix/store/75d0ra57gdcsmgqrrx42lrpvx97bvynb-bash-4.4-p23/bin/bash gsettingsschemadir=/nix/store/2qh38kp936vlb6pwrl2kba0q1gpc3091-surge-1.9.0/share/gsettings-schemas/surge-1.9.0/glib-2.0/schemas/ install
make: *** No rule to make target 'install'.  Stop.
builder for '/nix/store/dbvbz9li4180xlhjvz2j0vwkdxcczycx-surge-1.9.0.drv' failed with exit code 2
```

Comment (jonringer) [reply pkgs/applications/audio/surge/default.nix:37]: you can also do `makeTargets = [ "install-everything-global" ];`

Comment (magnetophon) [reply pkgs/applications/audio/surge/default.nix:37]: @jonringer Thanks. 

I can't find the cmake equivalent.
Is there one?

Comment (jonringer) [reply pkgs/applications/audio/surge/default.nix:37]: oh, not sure if there is one.

I think it's fine as is

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 123937 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>surge</li>  </ul></details>

**Pull request merged**